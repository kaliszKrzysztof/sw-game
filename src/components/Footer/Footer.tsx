import React from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      position: 'absolute',
      bottom: 0,
      width: '100%',
      display: 'flex',
      justifyContent: 'center',
      padding: theme.spacing(4, 0),
      borderTop: `1px solid ${theme.palette.divider}`,
      backgroundColor: theme.palette.grey[100],
    },
    swapiHeader: {
      fontSize: '2rem',
      color: '#FFE81F',
      fontWeight: theme.typography.fontWeightBold,
    },
    swapiText: {
      color: '#FFE81F',
      fontWeight: theme.typography.fontWeightMedium,
    },
  }),
);

const Footer: React.FC = () => {
  const classes = useStyles();

  return (
    <footer className={classes.root}>
      <div>
        <Typography align="center" color="textSecondary">
          Powered by
        </Typography>
        <Typography align="center" className={classes.swapiHeader}>
          SWAPI
        </Typography>
        <Typography align="center" color="textSecondary">
          The Star Wars API
        </Typography>
      </div>
    </footer>
  );
};

export default Footer;
