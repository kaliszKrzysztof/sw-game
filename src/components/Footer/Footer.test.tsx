import React from 'react';
import { render } from '@testing-library/react';
import Footer from './Footer';

describe('<Footer />', () => {
  it('should render', () => {
    render(<Footer />);
  });
  it('matches the snapshot', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
