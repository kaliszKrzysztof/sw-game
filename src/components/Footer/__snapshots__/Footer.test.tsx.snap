// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches the snapshot 1`] = `
<div>
  <footer
    class="makeStyles-root-4"
  >
    <div>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Powered by
      </p>
      <p
        class="MuiTypography-root makeStyles-swapiHeader-5 MuiTypography-body1 MuiTypography-alignCenter"
      >
        SWAPI
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        The Star Wars API
      </p>
    </div>
  </footer>
</div>
`;
